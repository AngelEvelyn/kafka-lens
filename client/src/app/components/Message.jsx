import PropTypes from 'prop-types';
import React from 'react';
import '../css/Message.scss';

const Message = ({ message, offset }) => (
  <span>
    <div className="single-message">
      {message}
      <p className="offset-indicator">{offset}</p>
    </div>
  </span>
);

export default Message;

Message.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  message: PropTypes.any.isRequired,
  offset: PropTypes.number.isRequired,
};
