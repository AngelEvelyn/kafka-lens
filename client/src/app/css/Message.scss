@import 'theme';

.single-message {
    // padding-top: 7px;
    // padding-bottom: 11px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-left: 20px;
    padding-right: 8px;
    min-height: 40px;
    line-height: 34px;
    color: lighten(#544d4d, 10%);
    border-bottom: 1px solid lighten($message-border-color, 2%);
}

.single-message p {
    margin: 0 3px;
}

.single-message:hover, .single-message:focus {
    background-color: #f7ae4f;
    color: white;
    cursor: pointer;
    box-shadow: 0 0 50px rgba(33,33,33,.2); 
    border-bottom: 1px solid $message-border-color;
}

.offset-indicator {
    color: $off-white-color;
    background-color: darken($highlight-bar-color, 10%);
    padding: 0 3px;
    // height: 100%;
    border-radius: 3px;
}

h6 {
    margin: 0;
}