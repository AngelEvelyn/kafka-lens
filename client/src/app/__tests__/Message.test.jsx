import React from 'react';
import { shallow } from 'enzyme';

import Message from '../components/Message';

describe('Message.jsx unit tests', () => {
  let wrapper;
  const props = {
    offset: 5,
    message: 'Test',
  };
  beforeAll(() => {
    wrapper = shallow(<Message {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render two p tags', () => {
    expect(wrapper.find('p')).toHaveLength(2);
  });

  it('Should render message-content with inner text "Test"', () => {
    expect(wrapper.find('.message-content').text()).toBe('Test');
  });

  it('Should render message-content with inner text "Test"', () => {
    expect(wrapper.find('.offset-indicator').text()).toBe('5');
  });
});
