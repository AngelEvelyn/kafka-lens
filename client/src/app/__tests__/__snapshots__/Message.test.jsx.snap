// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.jsx unit tests Should render without errors 1`] = `
<span>
  <div
    className="single-message"
  >
    <p
      className="message-content"
    >
      Test
    </p>
    <p
      className="offset-indicator"
    >
      5
    </p>
  </div>
</span>
`;
